openapi/cue/build.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/emicklei/proto"
+	"github.com/kr/pretty"
+)
+
+type plan map[string][]grouping
+
+// buildPlan specifies which .proto files end up in which OpenAPI .json files.
```

Comment:
Would it make sense to have this in a config file rather than baked into the tool?

Reply:
Sure, although @jasonwzm mentioned he preferred flags. That seems a bit tricky to me, as the all-at-once approach seems better to me, but that can work too.

Reply:
Imagine just consuming a .yaml file which lists this stuff and other related options. The command-line just specifies the path to the yaml file.

Reply:
Sure I can do that. @jasonwzm: would that work for you and that we eliminate the other command line flags, or would you still need these?

Reply:
Sure I can do that. @jasonwzm: would that work for you and that we eliminate the other command line flags, or would you still need these?

Reply:
I think either works fine. I think the preference is that the config file or command flags should be specified in the istio/api repo instead of in the tool itself. Thanks!